Add properties for Dynatrace metrics API v2 ingest with Micrometer (#1)
dynatrace-oss-contrib/spring-boot: spring-boot-project/spring-boot-docs/src/docs/asciidoc/production-ready-features.adoc

	        step: "30s"
----

[[production-ready-metrics-export-dynatrace]]
==== Dynatrace

Dynatrace offers two metrics ingest APIs:

===== API v1

> The Dynatrace API v1 registry pushes metrics to the configured URI periodically.

Reviewer: ```suggestion
The Dynatrace API v1 registry pushes metrics to the configured URI periodically using the https://www.dynatrace.com/support/help/dynatrace-api/environment-api/metric-v1/[Timeseries API v1].
```

	        uri: "YOUR_URI"
----

> You can also change the interval at which metrics are sent to Dynatrace:

Reviewer: I think it would make sense to leave the example setting "step" in for v1 so we don't alter the existing one.

> The API v2 can be used int two ways: If a local OneAgent is running on the machine, it is enough to tell the exporter to use v2 and metrics will be automatically exported to the https://www.dynatrace.com/support/help/how-to-use-dynatrace/metrics/metric-ingestion/ingestion-methods/local-api/[local OneAgent endpoint], which forwards them to the Dynatrace backend:

Reviewer: ```suggestion
The API v2 can be used int two ways:

If a local OneAgent is running on the machine, it is enough to tell the exporter to use v2 and metrics will be automatically exported to the https://www.dynatrace.com/support/help/how-to-use-dynatrace/metrics/metric-ingestion/ingestion-methods/local-api/[local OneAgent ingest endpoint], which forwards them to the Dynatrace backend:
```

[source,yaml,indent=0,configprops,configblocks]
----
	management:
	  metrics:
	    export:
	      dynatrace:
	        api-version: "v2"
----

> 

Reviewer: ```suggestion

Note that the [local OneAgent endpoint] will have to be enabled first as stated in the documentation.

```

> If no local OneAgent is running, the token and URI are required.

Reviewer: ```suggestion
If no local OneAgent is running, the endpoint of the https://www.dynatrace.com/support/help/dynatrace-api/environment-api/metric-v2/post-ingest-metrics/[Metrics API v2] and an API token are required.
```

> The https://www.dynatrace.com/support/help/dynatrace-api/basics/dynatrace-api-authentication/[API token] must have the `metrics.ingest` permission set.

Reviewer: ```suggestion
The https://www.dynatrace.com/support/help/dynatrace-api/basics/dynatrace-api-authentication/[API token] must have the "Ingest metrics" (`metrics.ingest`) permission set.
It is recommended to limit scope to only this one.
```

[source,yaml,indent=0,configprops,configblocks]
----
	management:
	  metrics:
	    export:
	      dynatrace:
	        api-version: "v2"
	        api-token: "YOUR_TOKEN"
	        uri: "YOUR_URI"
----

> When using the Dynatrace v2 API, new optional features are available:

Reviewer: ```suggestion
When using the Dynatrace v2 API, the following optional features are available:
```

> * Metric key prefix: set a prefix that will be prepended to all exported metric names.

Reviewer: ```suggestion
* Metric key prefix: Sets a prefix that will be prepended to all exported metric names.
```

* Enrich with OneAgent metadata: If a OneAgent is running, enrich metrics with information about the host and the process.

> * Default dimensions: Specify key-value pairs that are added to all exported metrics.

Reviewer: ```suggestion
* Default dimensions: Specify key-value pairs that are added to all exported metrics. If tags with the same key are specified using Micrometer, they overwrite the default dimensions.
```

[source,yaml,indent=0,configprops,configblocks]
----
	management:
	  metrics:
	    export:
	      dynatrace:

> 	      # ... token and uri or blank for OneAgent export, and api-version: "v2" ...

Reviewer: Odd indentation and api-version should not be commented out.

Author: this is just to tell users that they should use `apiVersion: v2`. I didnt put it here explicitly to reduce duplication but it makes sense to add it. I added it back.

	        api-version: "v2"
----

Note that the https://www.dynatrace.com/support/help/how-to-use-dynatrace/metrics/metric-ingestion/ingestion-methods/local-api/[local OneAgent endpoint] will have to be enabled first as stated in the documentation.

If no local OneAgent is running, the endpoint of the https://www.dynatrace.com/support/help/dynatrace-api/environment-api/metric-v2/post-ingest-metrics/[Metrics API v2] and an API token are required.
The https://www.dynatrace.com/support/help/dynatrace-api/basics/dynatrace-api-authentication/[API token] must have the "Ingest metrics" (`metrics.ingest`) permission set.

> It is recommended to limit scope to only this one.

Reviewer: ```suggestion
It is recommended to limit scope to only this one permission.
```
